test/suites/test_datetime_arithmetic.py

```diff
@@ -0,0 +1,317 @@
+        },
+        'nsecs': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+            'arg_2': tarantool.Interval(nsec=10000023),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43,
+                                      nsec=10000023),
+        },
+        'zero': {
+            'arg_1': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+            'arg_2': tarantool.Interval(),
+            'res': tarantool.Datetime(year=2008, month=2, day=3, hour=3, minute=36, sec=43),
+        },
+        'month_non_last_day_none_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=1, day=30),
+            'arg_2': tarantool.Interval(month=13, adjust=tarantool.IntervalAdjust.NONE),
+            'res': tarantool.Datetime(year=2010, month=2, day=28),
+        },
+        'month_non_last_day_excess_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=1, day=30),
+            'arg_2': tarantool.Interval(month=13, adjust=tarantool.IntervalAdjust.EXCESS),
+            'res': tarantool.Datetime(year=2010, month=3, day=2),
+        },
+        'month_non_last_day_last_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=3, day=30),
+            'arg_2': tarantool.Interval(month=2, adjust=tarantool.IntervalAdjust.LAST),
+            'res': tarantool.Datetime(year=2009, month=5, day=30),
+        },
+        'month_overflow_last_day_last_adjust': {
+            'arg_1': tarantool.Datetime(year=2009, month=2, day=28),
+            'arg_2': tarantool.Interval(month=1, adjust=tarantool.IntervalAdjust.LAST),
+            'res': tarantool.Datetime(year=2009, month=3, day=31),
+        },
+    }
+
+    def test_python_datetime_add(self):
+        for name in self.datetime_add_cases.keys():
+            with self.subTest(msg=name):
+                case = self.datetime_add_cases[name]
+
+                self.assertEqual(case['arg_1'] + case['arg_2'], case['res'])
+
+    @skip_or_run_datetime_test
+    def test_tarantool_datetime_add(self):
+        for name in self.datetime_add_cases.keys():
+            with self.subTest(msg=name):
+                case = self.datetime_add_cases[name]
+
+                self.assertSequenceEqual(self.con.call('add', case['arg_1'], case['arg_2']),
+                                         [case['res']])
+
+    datetime_add_winter_time_switch_case = {
+        'arg_1': tarantool.Datetime(year=2008, month=1, day=1, tz='Europe/Moscow'),
+        'arg_2': tarantool.Interval(month=6),
+        'res': tarantool.Datetime(year=2008, month=7, day=1, tz='Europe/Moscow'),
+    }
+
+    def test_python_datetime_add_winter_time_switch(self):
+        case = self.datetime_add_winter_time_switch_case
+
+        self.assertEqual(case['arg_1'] + case['arg_2'], case['res'])
+
+    @skip_or_run_datetime_test
+    @unittest.expectedFailure # See https://github.com/tarantool/tarantool/issues/7700
```

Comment:
Wil the testsl fail on a new versions of Tarantool with fixes? It seems better to skip the test.

Reply:
https://github.com/tarantool/tarantool-python/pull/230#discussion_r983429755